File: Ruby/first_non_repeat.rb
def first_non_repeat(string)
    map = {}
    splitted = string.split("")
    splitted.each do |i|
        temp = splitted.count(i)
        map[i] = temp
    end
    map.each {|k,v| return k if v==1}
    -1
end

p first_non_repeat("leetcode")

File: JavaScript/firstNonRepeat.js
let c = "leetcode"
let j = "aaa"


function firstNonRepeat(s){
    let map = {}
    for(const i of s){
        map[i] = (map[i] || 0) + 1
    }
    console.log(map)
    for(const j in map){
       if(map[j]===1) return s.indexOf(j)
    }
    return -1
}

console.log(firstNonRepeat(c))